src/software/ai/hl/stp/tactic/pass_defender/pass_defender_fsm.cpp

```diff
@@ -105,3 +106,50 @@ void PassDefenderFSM::interceptBall(const Update& event)
         TbotsProto::DribblerMode::MAX_FORCE, TbotsProto::BallCollisionType::ALLOW,
         AutoChipOrKick{AutoChipOrKickMode::OFF, 0}));
 }
+
+bool PassDefenderFSM::ballNearbyWithoutThreat(const Update& event)
+{
+    Point robot_position   = event.common.robot.position();
+    std::optional<Robot> nearest_enemy =
+            event.common.world_ptr->enemyTeam().getNearestRobot(robot_position);
+    if (nearest_enemy)
+    {
+        // Get the ball if ball is closer to robot than enemy threat by threshold ratio
+        double ball_distance =
+                distance(robot_position, event.common.world_ptr->ball().position());
+        double nearest_enemy_distance =
+                distance(robot_position, nearest_enemy->position());
+
+        LOG(VISUALIZE) << *createDebugShapes({
+                 *createDebugShape(
+                         Circle(robot_position,
+                                std::min(nearest_enemy_distance * MAX_GET_BALL_RATIO_THRESHOLD, MAX_GET_BALL_RADIUS_M)),
+                         std::to_string(event.common.robot.id()) + "1",
+                         "ballgetzone"
+                 )
+         });
+
+        bool ball_is_near_friendly = ball_distance < nearest_enemy_distance * MAX_GET_BALL_RATIO_THRESHOLD;
+        bool ball_is_within_max_range = ball_distance <= MAX_GET_BALL_RADIUS_M;
+        bool ball_is_slow = event.common.world_ptr->ball().velocity().length() <= MAX_BALL_SPEED_TO_GET_MS;
+
+        return  ball_is_near_friendly && ball_is_within_max_range && ball_is_slow;
+    }
+    else
+    {
+        return true;
+    }
```

Comment:
try to see if you can add a utility function that could be shared between the crease defenders and pass defenders for this logic

Reply:
Is there a way I can do this without inheritance? The two bits of logic are slightly different, and I am struggling to be able to define stuff like control params and Update outside of an FSM/Class

Reply:
Ykw, I will just get this cleaned up after both PRs are approved and merged in. Otherwise, I will have some weirdness between having duplicate implementations of the utility functions.

Reply:
Nima's suggestion with moving the core logic (no FSM stuff) somewhere to `software/ai/evaluation` is a good idea

Reply:
I ended up just making a `DefenderFSMBase` superclass since it makes a lot more sense rather than having a few loose functions in the eval folder.